src/main/java/com/facebook/openwifirrm/DeviceDataManager.java

```diff
@@ -106,7 +106,13 @@ private DeviceTopology readTopology(File topologyFile) throws IOException {
 			// Read file
 			logger.info("Reading topology file '{}'", topologyFile.getPath());
 			String contents = Utils.readFile(topologyFile);
-			topo = gson.fromJson(contents, DeviceTopology.class);
+			if (contents == null || contents.isBlank()) {
```

Comment:
- `Utils.readFile()` will never return null, don't check for it
- Use `.isEmpty()` instead of `.isBlank()`

However this whole block seems too verbose, you can write it much more succinctly (without the branching and duplicate logic):
```java
		DeviceTopology topo = null;
		if (topologyFile.isFile()) {
			// Read file
			logger.info("Reading topology file '{}'", topologyFile.getPath());
			String contents = Utils.readFile(topologyFile);
			topo = gson.fromJson(contents, DeviceTopology.class);
		}
		if (topo == null) {
			// Missing/empty file, write defaults to disk
			logger.info(
				"Creating default topology file '{}'", topologyFile.getPath()
			);
			topo = new DeviceTopology();
			Utils.writeJsonFile(topologyFile, topo);
		}
```
Note this is possible since `gson.fromJson()` properly handles empty strings and returns null, so you don't need to check it yourself.

Please do the same in the other methods too.

Reply:
Will do. This will result in the log not showing whether the file is missing vs blank, but I suppose this isn't a big deal.